base/src/commonMain/kotlin/utils/MeasurementUnit.kt

```diff
@@ -0,0 +1,320 @@
+
+object Hectometer : ScientificUnit.MetricLength("hm"), Serializable
+
+object Kilometer : ScientificUnit.MetricLength("km"), Serializable
+
+object Meter : ScientificUnit.MetricLength("m"), Serializable
+
+object Inch : ScientificUnit.ImperialLength("\'"), Serializable
+
+object Foot : ScientificUnit.ImperialLength("\""), Serializable
+
+object Yard : ScientificUnit.ImperialLength("yd"), Serializable
+
+object Mile : ScientificUnit.ImperialLength("mi"), Serializable
+
+object Celsius : ScientificUnit.MetricTemperature("°C"), Serializable
+
+object Kelvin : ScientificUnit.MetricTemperature("K"), Serializable
+
+object Fahrenheit : ScientificUnit.USImperialTemperature("°F"), Serializable
+
+object Tonne : ScientificUnit.MetricWeight("t"), Serializable
+
+object Kilogram : ScientificUnit.MetricWeight("kg"), Serializable
+
+object Gram : ScientificUnit.MetricWeight("g"), Serializable
+
+object Milligram : ScientificUnit.MetricWeight("mg"), Serializable
+
+object Microgram : ScientificUnit.MetricWeight("mcg"), Serializable
+
+object ImperialTon : ScientificUnit.UKImperialWeight("ton"), Serializable // also long ton
+
+object UsTon : ScientificUnit.USImperialWeight("ton"), Serializable // also short ton
+
+object Stone : ScientificUnit.ImperialWeight("st"), Serializable
+
+object Pound : ScientificUnit.ImperialWeight("lb"), Serializable
+
+object Ounce : ScientificUnit.ImperialWeight("oz"), Serializable
+
+object Dram : ScientificUnit.ImperialWeight("dr"), Serializable
+
+object Grain : ScientificUnit.ImperialWeight("gr"), Serializable
+
+object UsLiquidGallon : ScientificUnit.USImperialVolume("gal"), Serializable
+
+object UsLiquidQuart : ScientificUnit.USImperialVolume("qt"), Serializable
+
+object UsLiquidPint : ScientificUnit.USImperialVolume("pint"), Serializable
+
+object UsLegalCup : ScientificUnit.USImperialVolume("cup"), Serializable
+
+object UsFluidOunce : ScientificUnit.USImperialVolume("fl oz"), Serializable
+
+object UsFluidDram : ScientificUnit.USImperialVolume("fl dr"), Serializable
+
+object CubicMeter : ScientificUnit.MetricVolume("cu m"), Serializable
+
+object Liter : ScientificUnit.MetricVolume("l"), Serializable
+
+object Milliliter : ScientificUnit.MetricVolume("ml"), Serializable
+
+object ImperialGallon : ScientificUnit.UKImperialVolume("gal"), Serializable
+
+object ImperialQuart : ScientificUnit.UKImperialVolume("qt"), Serializable
+
+object ImperialPint : ScientificUnit.UKImperialVolume("pt"), Serializable
+
+object ImperialCup : ScientificUnit.UKImperialVolume("cup"), Serializable
+
+object ImperialFluidOunce : ScientificUnit.UKImperialVolume("fl oz"), Serializable
+
+object ImperialFluidDram : ScientificUnit.UKImperialVolume("fl dr"), Serializable
+
+object CubicFoot : ScientificUnit.UKImperialVolume("cu ft"), Serializable
+
+object CubicInch : ScientificUnit.UKImperialVolume("cu in"), Serializable
+
+private val MILLIMETERS_IN_METER = 1000.0.toDecimal()
```

Comment:
Instead of storing these values in here, you can store them in the objects describing their unit. so MILLIMETERS_IN_METER goes into object Millimeter

In addition, I would suggest to move toBaseSi / fromBaseSi into the ScientificUnit sealed class and provide implementation in the objects.

Reply:
I Agree -
"Instead of storing these values in here, you can store them in the objects describing their unit. soMILLIMETERS_IN_METER goes into object Millimeter"

Could you elaborate more on this? I am using toBaseSi / fromBaseSi internally just to do calculations and they are not meant to be exposed. I think also to do that the objects need to hold the actual value whereas we now use them as enums for type checks. 
"In addition, I would suggest to move toBaseSi / fromBaseSi into the ScientificUnit sealed class and provide implementation in the objects."